File: sdk/elixir/lib/rocksky/apikey.ex
defmodule Rocksky.Apikey do
  @moduledoc "`app.rocksky.apikey.*` endpoints. All require an authenticated client."

  alias Rocksky.HTTP

  @doc "List your API keys. Params: `:limit`, `:offset`."
  def get_apikeys(client, params \\ []),
    do: HTTP.query(client, "app.rocksky.apikey.getApikeys", params)

  @doc "Create a new API key. Body: `:name`, `:description`."
  def create_apikey(client, body) when is_list(body) or is_map(body),
    do: HTTP.procedure(client, "app.rocksky.apikey.createApikey", [], Map.new(body))

  @doc "Update an API key. Body: `:id`, `:name`, `:description`."
  def update_apikey(client, body) when is_list(body) or is_map(body),
    do: HTTP.procedure(client, "app.rocksky.apikey.updateApikey", [], Map.new(body))

  @doc "Remove an API key. Params: `:id`."
  def remove_apikey(client, params),
    do: HTTP.procedure(client, "app.rocksky.apikey.removeApikey", params)
end

File: sdk/clojure/src/rocksky/apikey.clj
(ns rocksky.apikey
  "Endpoints under app.rocksky.apikey.*"
  (:require [rocksky.client :as c]))

(defn get-apikeys
  "List API keys for the authenticated user.

  Optional: `:limit` `:offset`."
  ([client] (get-apikeys client nil))
  ([client {:keys [limit offset]}]
   (c/query client :app.rocksky.apikey.getApikeys
            {:limit limit :offset offset})))

(defn create-apikey
  "Create a new API key.

  Required: `:name`. Optional: `:description`."
  [client {:keys [name description] :as body}]
  (c/procedure client :app.rocksky.apikey.createApikey body))

(defn update-apikey
  "Update an existing API key.

  Required: `:id` `:name`. Optional: `:description`."
  [client {:keys [id name description] :as body}]
  (c/procedure client :app.rocksky.apikey.updateApikey body))

(defn remove-apikey
  "Remove an API key.

  Required: `:id`."
  [client {:keys [id]}]
  (c/procedure client :app.rocksky.apikey.removeApikey nil {:id id}))
